### PR TITLE
[Data Objects] introduce onContainerSaved method 

### LINKED ISSUE #5965: [Data Objects] introduce onContainerSaved method
Call when a container definition gets saved, i.e.
- object brick definition
- fieldcollection  definition
- class definition

Two methods:
- preSaved
- postSaved

Input params:
* container definition
* additional params array containing additional contextual information

classSaved method is currently only when the field resides inside a classdefinition but not for other container types.

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #5965

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Objectbricks.php [hunk 1]
```diff
@@ -1012,4 +1015,18 @@ public function denormalize($value, $params = [])
 
         return null;
     }
+
+    /** {@inheritdoc } */
+    public function preSave($containerDefinition, $params = [])
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Objectbricks.php]: why not just use `ClassSavedTrait`?

File: models/DataObject/ClassDefinition/Data/Objectbricks.php [hunk 2]
```diff
@@ -892,7 +893,9 @@ public function classSaved($class, $params = [])
                         if (method_exists($fd, 'classSaved')) {
                             if (!$fd instanceof Localizedfields) {
                                 // defer creation
-                                $fd->classSaved($class);
+                                if (!$fd instanceof DataContainerAwareInterface) {
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Objectbricks.php]: Localizedfields field check not required anymore as it implements `DataContainerAwareInterface`. just use one if condition for interface & method_exists check.

File: models/DataObject/ClassDefinition/Data/Fieldcollections.php [hunk 3]
```diff
@@ -691,7 +691,7 @@ public function classSaved($class, $params = [])
                     $fieldDefinition = $definition->getFieldDefinitions();
 
                     foreach ($fieldDefinition as $fd) {
-                        if (method_exists($fd, 'classSaved')) {
+                        if (!$fd instanceof DataContainerAwareInterface && method_exists($fd, 'classSaved')) {
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Fieldcollections.php:697]: Localizedfields field check not required

File: models/DataObject/ClassDefinition/Data/Fieldcollections.php [hunk 4]
```diff
@@ -922,4 +923,18 @@ public function denormalize($value, $params = [])
 
         return null;
     }
+
+    /** {@inheritdoc } */
+    public function preSave($containerDefinition, $params = [])
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Fieldcollections.php]: `ClassSavedTrait`?

**Review by dvesh3 — CHANGES_REQUESTED** [review]
error when using AdvanceManyToManyRelation inside Objectbrick:
image

AdvManyToManyRelations classSaved method never called on objectbrick save:
image

File: models/DataObject/ClassDefinition/Data/Objectbricks.php (hunk 1 above)
Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/Objectbricks.php]: introduced the trait afterwards and then missed this one :-) thx

**Pull request merged**